File: level_1/PYTHON/체육복.py
def solution(n, lost, reserve):
    answer = 0
    student={}

    for i in range(n):
        student[i+1]=1

    for i in lost:
        student[i]-=1
        
    for z in reserve:    
        student[z]+=1
        
    for k in student:
        if student[k]==0:
            if k != 1 and k != n:
                if student[k-1]>1:
                    student[k]+=1
                    student[k-1]-=1
                elif student[k+1]>1:
                    student[k]+=1
                    student[k+1]-=1
            elif k==1:
                if student[k+1]>1:
                    student[k]+=1
                    student[k+1]-=1
            elif k==n:
                if student[k-1]>1:
                    student[k]+=1
                    student[k-1]-=1        
                    

    for i in student.values():
        if i!=0:
            answer+=1
            
    return answer

File: level_1/C++/체육복.cpp
#include <vector>
#include <map>
#include <algorithm>
using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
	int answer = 0;
	map<int, int> student;

	for (int i = 1; i <= n; i++) {
		auto lost_itr = find(lost.begin(), lost.end(), i);
		auto reserve_itr = find(reserve.begin(), reserve.end(), i);

		student[i] = 1;

		if (lost_itr != lost.end()) {
			student[i]--;
		}
		if (reserve_itr != reserve.end()) {
			student[i]++;
		}
	}


	for (int i = 1; i <= n; i++) {
		if (i != 1 || i != n) {
			if (student[i] == 0 && student[i - 1] == 2) {
				student[i]++;
				student[i - 1]--;
			}
			else if (student[i] == 0 && student[i + 1] == 2) {
				student[i]++;
				student[i + 1]--;
			}
		}

		else if (i == 1) {
			if (student[i] == 0 && student[i + 1] == 2) {
				student[i]++;
				student[i + 1]--;
			}
		}
		else {
			if (student[i] == 0 && student[i - 1] == 2) {
				student[i]++;
				student[i - 1]--;
			}
		}
	}

	for (auto elem : student) {
		if (elem.second != 0) answer++;
	}

	return answer;
}